extensions/blocks/calendly/edit.js

```diff
@@ -0,0 +1,367 @@
+	}
+
+	if ( backgroundColor && backgroundColor.match( hexRegex ) ) {
+		attributes.backgroundColor = backgroundColor;
+	}
+
+	if ( primaryColor && primaryColor.match( hexRegex ) ) {
+		attributes.primaryColor = primaryColor;
+	}
+
+	if ( textColor && textColor.match( hexRegex ) ) {
+		attributes.textColor = textColor;
+	}
+
+	return getValidatedAttributes( attributeDetails, attributes );
+};
+
+export default function CalendlyEdit( { attributes, className, setAttributes } ) {
+	const validatedAttributes = getValidatedAttributes( attributeDetails, attributes );
+
+	if ( ! isEqual( validatedAttributes, attributes ) ) {
+		setAttributes( validatedAttributes );
+	}
+
+	const {
+		backgroundColor,
+		submitButtonText,
+		hideEventTypeDetails,
+		primaryColor,
+		textColor,
+		style,
+		url,
+	} = validatedAttributes;
+	const [ embedCode, setEmbedCode ] = useState();
+	const [ notice, setNotice ] = useState();
+
+	const setErrorNotice = () =>
+		setNotice(
+			<>
+				{ __(
+					"Your calendar couldn't be embedded. Please double check your URL or code.",
+					'jetpack'
+				) }
+			</>
+		);
+
+	const parseEmbedCode = event => {
+		if ( ! event ) {
+			setErrorNotice();
+			return;
+		}
+
+		event.preventDefault();
+
+		if ( ! embedCode ) {
+			setErrorNotice();
+			return;
+		}
+
+		const newUrlAndStyle = getUrlAndStyleFromEmbedCode( embedCode );
+		if ( ! newUrlAndStyle ) {
+			setErrorNotice();
+			return;
+		}
+
+		setAttributes( getNewAttributesFromUrl( newUrlAndStyle ) );
+	};
+
+	const embedCodeForm = (
+		<form onSubmit={ parseEmbedCode }>
+			<input
+				type="text"
+				id="embedCode"
+				onChange={ event => setEmbedCode( event.target.value ) }
+				placeholder={ __( 'Calendly web address or embed code…' ) }
+				value={ embedCode }
+				className="components-placeholder__input"
+			/>
+			<div>
+				<Button isLarge type="submit">
+					{ _x( 'Embed', 'button label', 'jetpack' ) }
+				</Button>
+			</div>
+			<p>
+				<ExternalLink
+					href="https://help.calendly.com/hc/en-us/articles/223147027-Embed-options-overview"
+					target="_blank"
+				>
+					{ __( 'Need help finding your embed code?' ) }
+				</ExternalLink>
+			</p>
+		</form>
+	);
+
+	const blockPlaceholder = (
+		<Placeholder
+			label={ __( 'Calendly', 'jetpack' ) }
+			instructions={ __( 'Enter your Calendly web address or embed code below.', 'jetpack' ) }
+			icon={ <BlockIcon icon={ icon } /> }
+			notices={
+				notice && (
+					<Notice status="error" isDismissible={ false }>
+						{ notice }
+					</Notice>
+				)
+			}
+		>
+			{ embedCodeForm }
+		</Placeholder>
+	);
+
+	const iframeSrc = () => {
+		let src = url + '?embed_domain=wordpress.com&amp;embed_type=Inline';
+		src += '&amp;hide_event_type_details=' + ( hideEventTypeDetails ? 1 : 0 );
+		src += '&amp;background_color=' + backgroundColor;
+		src += '&amp;primary_color=' + primaryColor;
+		src += '&amp;text_color=' + textColor;
```

Comment:
Do we have a team view on template literals versus string concatenation? There is a mix of both in the component (see line 209 just below). If there isn't a current team view on this, is it something we want to deciding on for consistency?  I would vote for template literals.

Reply:
imo we can go with template literals here in blocks. That's something I've been encouraging, but we have not made it a hard rule yet.